app/src/main/java/org/p2p/wallet/auth/gateway/repository/mapper/GatewayServiceCreateWalletMapper.kt

```diff
@@ -101,36 +111,45 @@ class GatewayServiceCreateWalletMapper(
         userPublicKey: Base58String,
         userPrivateKey: Base58String,
         etheriumAddress: String,
-        phoneNumber: String,
+        phoneNumber: PhoneNumber,
         jsonEncryptedMnemonicPhrase: JsonObject,
+        socialShareOwnerId: String,
+        userSeedPhrase: List<String>,
         thirdShare: Web3AuthSignUpResponse.ShareDetailsWithMeta,
         otpConfirmationCode: String
     ): GatewayServiceRequest<ConfirmRegisterWalletRequest> {
-
-        val encryptedPayloadStrJson = jsonEncryptedMnemonicPhrase.toString()
+        val encryptedMetadata: GatewayOnboardingMetadataCiphered = onboardingMetadataCipher.encryptMetadata(
+            mnemonicPhrase = userSeedPhrase,
+            onboardingMetadata = GatewayOnboardingMetadata(
+                deviceShareDeviceName = Build.MANUFACTURER + Build.PRODUCT,
+                customSharePhoneNumberE164 = phoneNumber.e164Formatted(),
+                socialShareOwnerEmail = socialShareOwnerId
+            )
+        )
 
         val signatureField: Base58String = signatureFieldGenerator.generateSignatureField(
             userPrivateKey = userPrivateKey,
             structToSerialize = ConfirmRegisterWalletSignatureStruct(
                 clientId = userPublicKey.base58Value,
                 etheriumId = etheriumAddress.lowercase(),
-                phone = phoneNumber,
-                encryptedShare = gson.toJson(thirdShare),
```

Comment:
"not a valid signature" returned from the gateway due to `gson.toJson` formatted JSON output